pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -410,3 +401,122 @@ func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStor
 		}
 	}
 }
+
+func fetchAndStoreGuestInfo(types libvirt.DomainGuestInfoTypes, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	domain, err := con.LookupDomainByName(domainName)
+	if err != nil {
+		log.Log.Errorf("Domain lookup failed: %v", err)
+		return
+	}
+	defer domain.Free()
+
+	guestInfo, err := domain.GetGuestInfo(types, 0)
+	if err != nil {
+		log.Log.Errorf("Fetching guest info failed: %v", err)
+		return
+	}
+
+	if len(guestInfo.Interfaces) != 0 {
```

Comment:
In these conditions, instead of checking the returned `guestInfo`, can you check the value of `types` parameter?
Then we would only update fields that user wants, and not fields that are returned.

Reply:
Yep done.